go.mod

```diff
@@ -1,8 +1,6 @@
 module github.com/cilium/ebpf
 
-go 1.23.0
-
-toolchain go1.24.1
+go 1.24
```

Comment:
This is what the toolchain now does when doing `go mod edit -go=1.24` (modulo the toolchain line). I hope this spells the end of the go version mess.